tempesta_fw/http_sess.c

```diff
@@ -1261,32 +1272,28 @@ tfw_cfg_op_jsch_parse_resp_code(TfwCfgSpec *cs, const char *val)
 }
 
 static void
-tfw_cfgop_jsch_set_delay_limit(TfwCfgSpec *cs, time_t limit)
+tfw_cfgop_jsch_set_delay_limit(TfwCfgSpec *cs)
 {
-	time_t min_limit = tfw_cfg_js_ch->delay_min + tfw_cfg_js_ch->delay_range;
-	time_t warn_limit = min_limit + HZ / 10; /* 0.1 sec */
-	const time_t def_limit = min_limit + HZ; /* 1 sec. */
+	const unsigned long min_limit	= tfw_cfg_js_ch->delay_min
+			+ msecs_to_jiffies(tfw_cfg_js_ch->delay_range);
+	const unsigned long warn_limit	= min_limit + msecs_to_jiffies(100);
+	const unsigned long def_limit	= min_limit + msecs_to_jiffies(1000);
 
-	if (!limit) {
+	if (!tfw_cfg_js_ch->delay_limit) {
 		tfw_cfg_js_ch->delay_limit = def_limit;
-
-		return;
 	}
-	if (limit < min_limit) {
+	else if (tfw_cfg_js_ch->delay_limit < min_limit) {
 		TFW_WARN_NL("%s: delay limit is too low, "
```

Comment:
Why not just exit with configuration error here? It seems that `min_limit` is too small value for `tfw_cfg_js_ch->delay_limit` anyway, and will lead to rejections of legitimate clients.

Reply:
That was a requirement of the JS Challenge task: https://github.com/tempesta-tech/tempesta/issues/536#issuecomment-343276205

>  Configuration process must enforce the value to be greater than delay_range + delay_min and show warning if it's less than delay_range + delay_min + 100. Default value is delay_range + delay_min + 1000 (enough ever fro cross atlantic connections and relatively slow hardware and software).

The last moment of time browser can send a request - `delay_range + delay_min`, and some time is required to deliver the request. `min_limit` adds extra 0.1s. Yes, this configuration is not very reliable: delivering via cross-Atlantic is slower, browsers can be slow, etc., while default limit (+1s) is reliable enough. But user may want to use less or more strict time lines and we should honour his choices.